android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiWebViewClient.java

```diff
@@ -97,6 +98,20 @@ public void onPageStarted(WebView view, String url, Bitmap favicon)
 		if (proxy == null || webView.hasSetUserAgent) {
 			return;
 		}
+		if (proxy.hasProperty(TiC.PROPERTY_ON_LINK)) {
+			Object onLink = proxy.getProperty(TiC.PROPERTY_ON_LINK);
+			if (onLink instanceof KrollFunction) {
+				KrollFunction onLinkFunction = (KrollFunction) onLink;
+				KrollDict args = new KrollDict();
+				args.put(TiC.EVENT_PROPERTY_URL, url);
+				Object result = onLinkFunction.call(proxy.getKrollObject(), args);
+				if (result == null || (result instanceof Boolean && ((Boolean) result) == false)) {
+					webView.stopLoading();
+					return;
+				}
+			}
+		}
+
```

Comment:
There's one issue here. The `onPageStarted()` method gets called after the HTTP request was sent. If you set up the URL to use a customer URL scheme that was intended to be overridden by `onlink`, then the WebView will display an "unknown url scheme" error page.

I'm able to reproduce this issue with the following:
```javascript
var window = Ti.UI.createWindow();
var webView = Ti.UI.createWebView({
	html: '<!DOCTYPE html><html><body><a href="myapp:HelloWorld">Tap this link</a></body></html>',
	onlink: function(e) {
		Ti.API.info("@@@ onlink() invoked. url: " + e.url);
		var endTime = new Date();
		endTime.setTime(endTime.getTime() + 1000);
		while (endTime >= new Date()) {}
		return (e.url !== "myapp:HelloWorld");
	},
});
window.add(webView);
window.open();
```

The above issue can be solved by using the [WebViewClient.shouldOverrideUrlLoading()](https://developer.android.com/reference/android/webkit/WebViewClient.html#shouldOverrideUrlLoading(android.webkit.WebView,%20java.lang.String)) method instead. But the downside to using this method (based on what I remember) is that it only gets called when tapping/clicking on links. It does **not** get called for URLs invoked in JavaScript within the webpage (and I know iOS' UIWebView does not have this limitation). But this might be okay since we've named the property `onlink`.